.github/workflows/Bonsai.yml

```diff
@@ -0,0 +1,460 @@
+      # Needed to attach files to releases
+      contents: write
+      # Needed to upload to GitHub Packages
+      packages: write
+    needs: [build-and-test, determine-changed-packages]
+    # Pushes always publish CI packages (configure-build.py will add the branch name to the version string for branches besides main)
+    # Published releases always publish packages
+    # A manual workflow only publishes packages if explicitly enabled
+    if: github.event_name == 'push' || github.event_name == 'release' || (github.event_name == 'workflow_dispatch' && github.event.inputs.will_publish_packages == 'true')
+    steps:
+      # ----------------------------------------------------------------------- Checkout
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: .github
+          sparse-checkout-cone-mode: false
+
+      # ----------------------------------------------------------------------- Setup tools
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 8.x
+
+      - name: Setup Python 3.11
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
+
+      # ----------------------------------------------------------------------- Download artifacts
+      - name: Download built packages
+        uses: actions/download-artifact@v4
+        with:
+          name: Packages
+          path: Packages
+
+      - name: Download release manifest
+        uses: actions/download-artifact@v4
+        with:
+          name: ReleaseManifest
+
+      - name: Download portable zip
+        uses: actions/download-artifact@v4
+        if: github.event_name == 'release'
+        with:
+          name: PortableZip
+
+      - name: Download installer
+        uses: actions/download-artifact@v4
+        if: github.event_name == 'release'
+        with:
+          name: Installer
+
+      # ----------------------------------------------------------------------- Filter NuGet packages
+      - name: Filter NuGet packages
+        run: python .github/workflows/filter-release-packages.py ReleaseManifest Packages
+
+      # ----------------------------------------------------------------------- Upload release assets
+      - name: Upload release assets
+        if: github.event_name == 'release'
+        run: gh release upload ${{github.event.release.tag_name}} Bonsai.zip Bonsai-*.exe --clobber
+        env:
+          GH_TOKEN: ${{github.token}}
+
+      # ----------------------------------------------------------------------- Push to GitHub Packages
+      - name: Push to GitHub Packages
+        run: dotnet nuget push "Packages/*.nupkg" --skip-duplicate --no-symbols --api-key ${{secrets.GITHUB_TOKEN}} --source https://nuget.pkg.github.com/${{github.repository_owner}}
+        env:
+          # This is a workaround for https://github.com/NuGet/Home/issues/9775
+          DOTNET_SYSTEM_NET_HTTP_USESOCKETSHTTPHANDLER: 0
```

Comment:
We talked about this during our meeting, but for the sake of posterity:

I've always specified `--api-key` here and still had this issue in the past.

The issue this is working around manifests itself as random intermittent failure of pushing packages, which is extraordinarily frustrating since it happens so late in the workflow.

In retrospect I should've taken it out when I was developing the workflow to see if it was truly necessary anymore, but I'd lean towards just leaving it.